### PR TITLE
FM2-351: Add tags to distinguish OpenMRS Encounters from Visits

### PR DESCRIPTION
**Pull request by Medhavi-16**
…s an OpenMRS Encounter or Visit

## Issue I worked on

see https://issues.openmrs.org/browse/FM2-351

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

**Comment by Medhavi-16** [comment]
@ibacher I added the tags to Encounter and Visit resource types. PTAL.

**Comment by theanandankit** [comment]
@Medhavi-16 thanks, looks good. @ibacher

**Review by varung-31 — APPROVED** [review]
Please rename the PR to something shorter like `Add tags to distinguish OpenMRS Encounters from Visits`

**Comment by Medhavi-16** [comment]
> Please rename the PR to something shorter like `Add tags to distinguish OpenMRS Encounters from Visits`

@varung-31 I renamed the PR.

File: api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImplTest.java [hunk 1]
```diff
@@ -463,4 +465,16 @@ public void toFhirResource_shouldTranslateEncounterTypeToEncounterTypeField() {
...
+	@Test
+	public void toFhirResource_shouldHaveEncounterTag() {
+		Encounter result = encounterTranslator.toFhirResource(omrsEncounter);
+		
+		assertThat(result, notNullValue());
+		assertThat(result.getMeta().getTag(), notNullValue());
+		assertThat(result.getMeta().getTag().get(0).getSystem(), equalTo(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG));
+		assertThat(result.getMeta().getTag().get(0).getCode(), equalTo(ENCOUNTER_TYPE_CODE));
+		assertThat(result.getMeta().getTag().get(0).getDisplay(), equalTo(TYPE_DISPLAY));
+
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImplTest.java:478]:
```suggestion
```

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java [hunk 2]
```diff
@@ -76,6 +77,8 @@ public Encounter toFhirResource(@Nonnull org.openmrs.Encounter openMrsEncounter)
 			    Collections.singletonList(encounterLocationTranslator.toFhirResource(openMrsEncounter.getLocation())));
 		}
 		
+		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "Encounter", "encounter-type-display");
+
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java:81]:
```suggestion
```

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java [hunk 3]
```diff
@@ -59,6 +60,9 @@ public Encounter toFhirResource(@Nonnull Visit visit) {
 		if (visit.getLocation() != null) {
 			encounterLocationTranslator.toFhirResource(visit.getLocation());
 		}
+		
+		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "Visit", "encounter-type-display");
+		
 		encounter.setClass_(mapLocationToClass(visit.getLocation()));
 		
 		encounter.getMeta().setLastUpdated(visit.getDateChanged());
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java:68]:
```suggestion

		encounter.setClass_(mapLocationToClass(visit.getLocation()));
		
		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "Visit", "encounter-type-display");
		encounter.getMeta().setLastUpdated(visit.getDateChanged());
```

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java [hunk 4]
```diff
@@ -59,8 +60,10 @@ public Encounter toFhirResource(@Nonnull Visit visit) {
 		if (visit.getLocation() != null) {
 			encounterLocationTranslator.toFhirResource(visit.getLocation());
 		}
+
 		encounter.setClass_(mapLocationToClass(visit.getLocation()));
 		
+		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "Visit", "encounter-type-display");
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java:66]: The tag itself should probably be "visit" or "encounter"; the display property should be the capitalised version of them, e.g. "Visit" or "Encounter" (`display` is usually what gets displayed to the user).

**Review by ibacher — CHANGES_REQUESTED** [review]

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java (hunk 4 above)
Comment (Medhavi-16) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java:66]: @ibacher Will change it.

Comment (Medhavi-16) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java:66]: @ibacher will the code value remain the same?

**Comment by Medhavi-16** [comment]
@ibacher I've made the changes. PTAL

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java (hunk 4 above)
Comment (ibacher) [reply api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java]: @Medhavi-16 I'm not quite sure I understand the question, but I think this line should read:

```java
encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "visit", "Visit");
```

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java [hunk 5]
```diff
@@ -76,6 +77,7 @@ public Encounter toFhirResource(@Nonnull org.openmrs.Encounter openMrsEncounter)
 			    Collections.singletonList(encounterLocationTranslator.toFhirResource(openMrsEncounter.getLocation())));
 		}
 		
+		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "Encounter", "Encounter");
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImpl.java:80]:
```suggestion
		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "encounter", "Encounter");
```

File: api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java [hunk 6]
```diff
@@ -59,8 +60,9 @@ public Encounter toFhirResource(@Nonnull Visit visit) {
 		if (visit.getLocation() != null) {
 			encounterLocationTranslator.toFhirResource(visit.getLocation());
 		}
+
 		encounter.setClass_(mapLocationToClass(visit.getLocation()));
-		
+		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "Visit", "Visit");
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImpl.java:65]:
```suggestion
		encounter.getMeta().addTag(FhirConstants.OPENMRS_FHIR_EXT_ENCOUNTER_TAG, "visit", "Visit");
```

File: api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImplTest.java [hunk 7]
```diff
@@ -83,7 +83,9 @@
 	
 	private static final String TYPE_CODE = "encounter-type-code";
 	
-	private static final String TYPE_DISPLAY = "encounter-type-display";
+	private static final String TYPE_DISPLAY = "Encounter";
+	
+	private static final String ENCOUNTER_TYPE_CODE = "Encounter";
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/EncounterTranslatorImplTest.java:88]:
```suggestion
	private static final String ENCOUNTER_TYPE_CODE = "encounter";
```

Shouldn't this be `TYPE_CODE` above?

File: api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImplTest.java [hunk 8]
```diff
@@ -70,7 +70,9 @@
 	
 	private static final String TYPE_CODE = "encounter-type-code";
 	
-	private static final String TYPE_DISPLAY = "encounter-type-display";
+	private static final String TYPE_DISPLAY = "Visit";
+	
+	private static final String VISIT_TYPE_CODE = "Visit";
```
Comment (ibacher) [inline api/src/test/java/org/openmrs/module/fhir2/api/translators/impl/VisitTranslatorImplTest.java:75]:
```suggestion
	private static final String VISIT_TYPE_CODE = "visit";
```

Shouldn't this be the value of `TYPE_CODE` above?

**Comment by Medhavi-16** [comment]
@ibacher thank you. I'll change the encounter and visit code value.

**Comment by Medhavi-16** [comment]
@ibacher I made the said changes. PTAL.

**Review by ibacher — APPROVED** [review]

**Pull request merged**